src/transformStateWithClones.js

```diff
@@ -7,7 +7,32 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const eachAction = [];
+  let initialState = { ...state };
+
+  for (const action of actions) {
+    let newState = { ...initialState };
+
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(newState, action.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const key of action.keysToRemove) {
+          delete newState[key];
+        }
+        break;
+
+      case 'clear':
+        newState = {};
+        break;
+    }
```

Comment:
[CODE STYLE]: It's recommended to include a `default` case in your switch statement to handle any unexpected `action.type` values. This can help with debugging and ensuring that your function behaves correctly even if it receives an unexpected action.